src/core/meshcop/dataset_manager_ftd.cpp

```diff
@@ -154,22 +154,30 @@ Error DatasetManager::HandleSet(Coap::Message &aMessage, const Ip6::MessageInfo
     // check network key
     if (Tlv::Find<NetworkKeyTlv>(aMessage, networkKey) == kErrorNone)
```

Comment:
Same situation here. The change to definition of `NetworkKey` (i.e. adding new data members) will cause issue here.

`Tlv::Find<NetworkKeyTlv>(aMessage, networkKey)` would require the TLV length read from `aMessage` to be at least `sizeof(networkKey)` (which now has additional fields). So a TLV which contains just the key won't be accepted as a valid TLV.

Here is version of `Tlv::Find<>` we are using here:
https://github.com/openthread/openthread/blob/b42261f88c6c8360cb8738118c0767f886d76110/src/core/common/tlvs.hpp#L321-L346